src/node_api.cc

```diff
@@ -658,9 +658,16 @@ void napi_clear_last_error(napi_env env) {
 napi_status napi_set_last_error(napi_env env, napi_status error_code,
                                 uint32_t engine_error_code,
                                 void* engine_reserved) {
-  env->last_error.error_code = error_code;
-  env->last_error.engine_error_code = engine_error_code;
-  env->last_error.engine_reserved = engine_reserved;
+
+  if (env == nullptr) {
+    if (error_code == napi_ok) {
```

Comment:
There's actually a superfluous `CHECK_ARG(env, env)` in `napi_get_value_double()` because that function only uses `env` to pass it to `RETURN_STATUS_IF_FALSE()` and nothing else, and `RETURN_STATUS_IF_FALSE()` is capable of dealing with a `nullptr` `env`. ... but I'll leave that in.